test/VaultRegistry.test.ts

```diff
@@ -1,445 +1,943 @@
+                await this.anotherERC165Mock.allowInterfaceId(
+                    VAULT_INTERFACE_ID
+                );
+                this.subject
+                    .connect(this.allowedRegisterVaultSigner)
+                    .registerVault(
+                        this.anotherERC165Mock.address,
+                        this.ownerSigner.address
+                    );
+                expect(await this.subject.vaults()).to.deep.equal([
+                    this.erc165Mock.address,
+                    this.anotherERC165Mock.address,
+                ]);
+            });
+
+            describe("access control:", () => {
+                it("allowed: any address", async () => {
+                    await withSigner(randomAddress(), async (s) => {
+                        await expect(this.subject.connect(s).vaults()).to.not.be
+                            .reverted;
+                    });
+                });
+            });
+        });
+
+        describe("#vaultForNft", () => {
+            it("resolves Vault address by VaultRegistry NFT", async () => {
+                expect(await this.subject.vaultForNft(this.nft)).to.equal(
+                    this.erc165Mock.address
+                );
+                expect(await this.subject.vaultForNft(this.randomNft)).to.equal(
+                    ethers.constants.AddressZero
+                );
+                await this.anotherERC165Mock.allowInterfaceId(
+                    VAULT_INTERFACE_ID
+                );
+                this.randomNft = Number(
+                    await this.subject
+                        .connect(this.allowedRegisterVaultSigner)
+                        .callStatic.registerVault(
+                            this.anotherERC165Mock.address,
+                            this.ownerSigner.address
+                        )
+                );
+                this.subject
+                    .connect(this.allowedRegisterVaultSigner)
+                    .registerVault(
+                        this.anotherERC165Mock.address,
+                        this.ownerSigner.address
+                    );
+                expect(await this.subject.vaultForNft(this.randomNft)).to.equal(
+                    this.anotherERC165Mock.address
+                );
+            });
+
+            describe("access control:", () => {
+                it("allowed: any address", async () => {
+                    await withSigner(randomAddress(), async (s) => {
+                        await expect(
+                            this.subject.connect(s).vaultForNft(this.nft)
+                        ).to.not.be.reverted;
+                    });
+                });
+            });
+
+            describe("edge cases", () => {
+                describe("when Vault NFT is not registered in VaultRegistry", () => {
+                    it("returns zero address", async () => {
+                        let randomNumber = Math.round(Math.random() * 100 + 10);
```

Comment:
I thought that was the purpose of having randomNft - not to regenerate this each time